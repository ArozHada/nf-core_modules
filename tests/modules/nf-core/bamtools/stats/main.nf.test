nextflow_process {

    name "Test Process BAMTOOLS_STATS"
    script "modules/nf-core/bamtools/stats/main.nf"
    process "BAMTOOLS_STATS"
    config "tests/modules/nf-core/bamtools/stats/nextflow.config"
    tag "bamtools_stats"

    test("Sarscov2 Illumina PE Sorted BAM") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                            [ id:'test', single_end:false ], // meta map
                            file(params.test_data['sarscov2']['illumina']['test_paired_end_bam'], checkIfExists: true)
                           ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(path(params.outdir).list()).match() }
            )
        }

    }

}
