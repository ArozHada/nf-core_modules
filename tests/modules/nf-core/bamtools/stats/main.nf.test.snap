{
    "Sarscov2 Illumina PE Sorted BAM": {
        "content": [
            [
                "test.bam.stats:md5,b7c3f01682673a652d664a2889b15b66",
                "versions.yml:md5,fc7e45728bb7d5d4a80ab217553f0172"
            ]
        ],
        "timestamp": "2023-04-07T23:19:37+0000"
    }
}